packages/components/date-picker/__tests__/date-picker.test.tsx

```diff
@@ -600,6 +601,48 @@ describe("DatePicker", () => {
       expect(onHeaderExpandedChangeSpy).not.toHaveBeenCalled();
     });
 
+    it("should focus the month and year when pressed in the picker", () => {
```

Comment:
I’m unclear about what you want to verify with this test. I think the issue is that the focus indicator misaligns when the up and down arrow keys are held down. We should verify that it is synchronized properly.

Reply:
Sorry, forgot to check the keyboard inputs as well. I've gone round the UI one more time, testing:

- mouse
- keyboard individual keystrokes
- held down keys

The current behaviour seems to be:
- mouse: blue focus and grey highlighter always in sync throughout interactions
- individual keystrokes: same
- HELD KEYS: the blue focus indicator always syncs to the focused date, but _sometimes_ it manages to travel beyond the grey highlighter, but will immediately return on the next **individual** keystroke.

This is from my perspective a big improvement, because the toggle on/off focus mutation is gone, and the only time the UI desyncs is _(sometimes)_ when you hold down a key.